lib2/client/okta.go

```diff
@@ -0,0 +1,667 @@
+// Client for making requests to Okta APIs
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/net/publicsuffix"
+
+	"github.com/99designs/keyring"
+	"github.com/segmentio/aws-okta/lib"
+	"github.com/segmentio/aws-okta/lib/mfa"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	OktaServerUs      = "okta.com"
+	OktaServerEmea    = "okta-emea.com"
+	OktaServerPreview = "oktapreview.com"
+	OktaServerDefault = OktaServerUs
+
+	// deprecated; use OktaServerUs
+	OktaServer = OktaServerUs
+
+	Timeout = time.Duration(60 * time.Second)
+)
+
+type OktaUser struct {
```

Comment:
This is just used for deserialization, right? Just internally to `client`? Consider unexporting